lib/improver/spotdata/neighbour_finding.py

```diff
@@ -526,10 +526,21 @@ def process(self, sites, orography, land_mask):
                          vertical_displacements), axis=1)
         data = np.expand_dims(data, 1).astype(np.float32)
 
+        # Regardless of input sitelist coordinate system, the site coordinates
+        # are stored as latitudes and longitudes in the neighbour cube.
+        if self.site_coordinate_system != ccrs.PlateCarree():
+            lon_lats = self._transform_sites_coordinate_system(
+                site_x_coords, site_y_coords, ccrs.PlateCarree())
+            longitudes = lon_lats[:, 0]
+            latitudes = lon_lats[:, 1]
+        else:
+            longitudes = site_x_coords
+            latitudes = site_y_coords
+
         # Create a cube of neighbours
         neighbour_cube = build_spotdata_cube(
             data, 'grid_neighbours', 1, site_altitudes.astype(np.float32),
-            site_y_coords.astype(np.float32), site_x_coords.astype(np.float32),
+            latitudes.astype(np.float32), longitudes.astype(np.float32),
             wmo_ids, neighbour_methods=[method_name],
             grid_attributes=['x_index', 'y_index', 'vertical_displacement'])
```

Comment:
None of this section is covered by a unit-test (like 45% of this module according to "coverage"). Please could you add something, possibly to existing tests.

Reply:
I think my earlier comment on the coverage was incorrect. Today I find 100% coverage for this module. :-)